### DISCUSSION TITLE
Revalidating a single loader (for realtime)

### DISCUSSION THREAD

**Discussion by barbinbrad** [remix-run/remix#5639] [Proposals]
In the supabase subscriptions example, we have something like this:

```tsx
useSubscription(
    () => {
      fetcher.submit(null, { method: "post" });
    },
    { event: "INSERT", table: "clicks" }
  );
```

I believe that this is going to refetch all routes when a "click" is inserted. I think `useRevalidator` would do the same. 

In the example, this is no problem -- but at scale -- this is a becomes a little more expensive. Sure we can define all types of business logic inside of `shouldRevalidate` -- but what would be really nice is if we had a mechanism to revalidate one route (like an action can post to one route). For me the really beautiful part about the `loader` is that it defines the "data requirements" of a route. If we could combine subscriptions with targeted revalidators, I think we could create some really nice UXs that aren't too heavy on the data fetching.

Here's what I was thinking:

```tsx
const revalidator = useRevalidateRoute('/clicks');
useSubscription(
    () => {
      revalidator.revalidate();
    },
    { event: "INSERT", table: "clicks" }
  );
```

**Comment by nmackey** [comment]
I have an ask for something that I think is similar or may help solve this use case.

I would like an additional param added to `shouldRevalidate` that let's use know if it is being called as a result of the `revalidate` function from `useRevalidator` being triggered.

I have loaders that I only want to trigger when `revalidate` is called.

I'm currently achieving this by comparing `current` and `next` URLs and making sure they are the same and the rest of the params are undefined, however it seems like there should be a param that tells me `shouldRevalidate` is being triggered by the `revalidate` function.

**Comment by punkpeye** [comment]
@barbinbrad Have you ever found a solution for this?

**Comment by brophdawg11** [comment]
This discussion is being closed as part of the Remix v3 repository cleanup.

Remix v1/v2 was [upstreamed into React Router v7](https://remix.run/blog/wake-up-remix), so if this proposal is still relevant, please reopen or continue the discussion in the [React Router](https://github.com/remix-run/react-router) repository.
